app/js/lib/foxbox/network.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+
+// Private members.
+const p = {
+  // Private properties.
+  // Whether we can connect to the box via a local connection.
+  local: Symbol('local'),
+  // Whether we can connect to the box via a remote connection.
+  remote: Symbol('remote'),
+  // A reference to the interval to get the online status.
+  pingInterval: Symbol('pingInterval'),
+
+  // Private methods.
+  fetch: Symbol('fetch'),
+  ping: Symbol('ping'),
+  pingBox: Symbol('pingBox')
+};
+
+export default class Network {
+  constructor(settings) {
+    this.settings = settings;
+
+    this[p.local] = false;
+    this[p.remote] = false;
+    this[p.pingInterval] = null;
+  }
+
+  /**
+   * Attach event listeners related to the connection status.
+   *
+   * @return {Promise}
```

Comment:
I was thinking it's good to have the same signature for all `init()` functions in the lib, but I agree it's pretty useless here.